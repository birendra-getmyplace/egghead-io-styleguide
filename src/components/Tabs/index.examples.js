import React from 'react'
import {storiesOf} from '@kadira/storybook'
import {lorem} from 'faker'
import Tabs from '.'

storiesOf('Tabs', module)
  .add('default', () => (
    <Tabs groups={[
      {
        title: lorem.words(),
        component: <div>{lorem.paragraph()}</div>,
      },
      {
        title: lorem.words(),
        component: <div>{lorem.paragraph()}</div>,
      },
    ]} />
  ))
