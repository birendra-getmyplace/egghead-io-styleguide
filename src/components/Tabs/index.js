import React, {Component, PropTypes} from 'react'
import {
  Tab as LibraryTab,
  Tabs as LibraryTabs,
  TabList as LibraryTabList,
  TabPanel as LibraryTabPanel,
} from 'react-tabs';
LibraryTabs.setUseDefaultStyles(false)
import {map} from 'lodash'

export default class Tabs extends Component {

  static propTypes = {
    groups: PropTypes.arrayOf(React.PropTypes.shape({
      title: PropTypes.string.isRequired,
      component: PropTypes.node.isRequired,
    })).isRequired,
  }

  state = {
    selected: 0,
  }

  handleSelect = (index) => (
    this.setState({
      selected: index,
    })
  )

  render() {
    const {groups} = this.props

    return (
      <LibraryTabs onSelect={this.handleSelect}>

        <LibraryTabList className='list pa0 ma0 bg-moon-gray flex-ns br2 br--top'>
          {map(groups, (group, index) => (
            <LibraryTab
              key={index}
              className={`
                f6
                pv3 ph4 ph3-ns
                ttu
                pointer
                bl bl-0-ns bb-ns bw2 bw1-ns
                ${this.state.selected === index
                  ? 'b--orange black'
                  : 'b--transparent black-40'
                }
              `}
            >
              {group.title}
            </LibraryTab>
          ))}
        </LibraryTabList>

        {map(groups, (group, index) => (
          <LibraryTabPanel 
            key={index}
            className='pa3'
          >
            {group.component}
          </LibraryTabPanel>
        ))}  

      </LibraryTabs>
    )
  }
}
