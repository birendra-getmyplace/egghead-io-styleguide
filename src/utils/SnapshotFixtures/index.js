import React from 'react'

export const StringFixture = 'String'

export const UrlFixture = 'https://google.com'

export const ImageUrlFixture = 'https://s3.amazonaws.com/uifaces/faces/twitter/kimcool/128.jpg'

export const ComponentFixture = () => (
  <div>
    Component
  </div>
)
