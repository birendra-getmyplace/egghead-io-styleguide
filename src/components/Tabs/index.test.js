import React from 'react'
import {create} from 'react-test-renderer'
import {StringFixture, ComponentFixture} from '../../utils/SnapshotFixtures'
import Tabs from '.'

test('default', () => (
  expect(create(
    <Tabs groups={[
      {
        title: StringFixture,
        component: <ComponentFixture />,
      },
      {
        title: StringFixture,
        component: <ComponentFixture />,
      },
    ]} />
  )).toMatchSnapshot()
))
