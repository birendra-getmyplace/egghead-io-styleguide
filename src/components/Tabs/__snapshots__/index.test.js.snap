exports[`test default 1`] = `
<div
  className="ReactTabs react-tabs"
  data-tabs={true}
  onClick={[Function]}
  onKeyDown={[Function]}>
  <ul
    className="ReactTabs__TabList list pa0 ma0 bg-moon-gray flex-ns br2 br--top"
    role="tablist">
    <li
      aria-controls="react-tabs-1"
      aria-disabled="false"
      aria-selected="true"
      className="ReactTabs__Tab 
                    f6
                    pv3 ph4 ph3-ns
                    ttu
                    pointer
                    bl bl-0-ns bb-ns bw2 bw1-ns
                    b--orange black
                   ReactTabs__Tab--selected"
      id="react-tabs-0"
      role="tab"
      tabIndex="0">
      String
    </li>
    <li
      aria-controls="react-tabs-3"
      aria-disabled="false"
      aria-selected="false"
      className="ReactTabs__Tab 
                    f6
                    pv3 ph4 ph3-ns
                    ttu
                    pointer
                    bl bl-0-ns bb-ns bw2 bw1-ns
                    b--transparent black-40
                  "
      id="react-tabs-2"
      role="tab"
      tabIndex={null}>
      String
    </li>
  </ul>
  <div
    aria-labelledby="react-tabs-0"
    className="ReactTabs__TabPanel pa3 ReactTabs__TabPanel--selected"
    id="react-tabs-1"
    role="tabpanel"
    style={
      Object {
        "display": null,
      }
    }>
    <div>
      Component
    </div>
  </div>
  <div
    aria-labelledby="react-tabs-2"
    className="ReactTabs__TabPanel pa3"
    id="react-tabs-3"
    role="tabpanel"
    style={
      Object {
        "display": "none",
      }
    } />
</div>
`;
